recipes/android-ndk/all/conanfile.py

```diff
@@ -0,0 +1,208 @@
+from conans import ConanFile, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+
+
+class AndroidNDKInstallerConan(ConanFile):
+    name = "android-ndk"
+    description = "The Android NDK is a toolset that lets you implement parts of your app in " \
+                  "native code, using languages such as C and C++"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://developer.android.com/ndk/"
+    topics = ("NDK", "android", "toolchain", "compiler")
+    license = "Apache-2.0"
+    short_paths = True
+    no_copy_source = True
+    exports_sources = ["cmake-wrapper.cmd", "cmake-wrapper"]
+
+    settings = {"os": ["Windows", "Linux", "Macos"],
+                "arch": ["x86_64"]}
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def configure(self):
+        if self.settings.arch != 'x86_64':
+            raise ConanInvalidConfiguration("No binaries available for other than 'x86_64' architectures")
+
+    def source(self):
+        tarballs = self.conan_data["sources"][self.version]["url"]
+        tools.get(**tarballs[str(self.settings.os)])
+        extracted_dir = self.name + "-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def build(self):
+        pass # no build, but please also no warnings
+
+    # from here on, everything is assumed to run in 2 profile mode, when using the ndk as a build requirement
+
+    @property
+    def _platform(self):
+        return {"Windows": "windows",
+                "Macos": "darwin",
+                "Linux": "linux"}.get(str(self.settings_build.os))
+
+    @property
+    def _android_abi(self):
+        return {"x86": "x86",
+                "x86_64": "x86_64",
+                "armv7": "armeabi-v7a",
+                "armv8": "arm64-v8a"}.get(str(self.settings_target.arch))
+
+    @property
+    def _llvm_triplet(self):
+        arch = {'armv7': 'arm',
+                'armv8': 'aarch64',
+                'x86': 'i686',
+                'x86_64': 'x86_64'}.get(str(self.settings_target.arch))
+        abi = 'androideabi' if self.settings_target.arch == 'armv7' else 'android'
+        return '%s-linux-%s' % (arch, abi)
+
+    @property
+    def _clang_triplet(self):
+        arch = {'armv7': 'armv7a',
+                'armv8': 'aarch64',
+                'x86': 'i686',
+                'x86_64': 'x86_64'}.get(str(self.settings_target.arch))
+        abi = 'androideabi' if self.settings_target.arch == 'armv7' else 'android'
+        return '%s-linux-%s' % (arch, abi)
```

Comment:
are these not identical 🤔 could there be one triplet?

Reply:
afaik, there is not armv7a for the llvm tripplet
https://llvm.org/doxygen/classllvm_1_1Triple.html

Reply:
ahhh thank you I missed that!